### PR TITLE
feat: Edit report name - Maximum character limit reached changes for Spender+Approver with code coverage

### PR DESCRIPTION
**Pull request by Dimple16**
### <samp>🤖 Generated by Copilot at 77de32c</samp>

Added validation and error handling for report name length in the edit report name popover component. This prevents users from entering report names that are too long and exceed the backend limit of 250 characters.

### <samp>🤖 Generated by Copilot at 77de32c</samp>

> _To edit a report name, you pop over_
> _But don't make it too long, or it's over_
> _The `Save` button is `disabled`_
> _And an error is displayed_
> _If the name length exceeds two-fifty, no clover_

### <samp>🤖 Generated by Copilot at 77de32c</samp>

*  Validate and limit report name length to 250 characters ([link](https://github.com/fylein/fyle-mobile-app/pull/2276/files?diff=unified&w=0#diff-77c23b66e77777536c08734faf41242078eec3c5e0f21ea26a9c32e0e7c87e34L13-R13), [link](https://github.com/fylein/fyle-mobile-app/pull/2276/files?diff=unified&w=0#diff-77c23b66e77777536c08734faf41242078eec3c5e0f21ea26a9c32e0e7c87e34L28-R37), [link](https://github.com/fylein/fyle-mobile-app/pull/2276/files?diff=unified&w=0#diff-8ca51b8c05b80a7daebe2b1db9af16f87699ee281eb30ac6264a1a35276aecafR38-R41), [link](https://github.com/fylein/fyle-mobile-app/pull/2276/files?diff=unified&w=0#diff-8ca51b8c05b80a7daebe2b1db9af16f87699ee281eb30ac6264a1a35276aecafR52-R56))
  - Add a disabled attribute to the Save button that checks if the report name length is greater than or equal to 250 ([link](https://github.com/fylein/fyle-mobile-app/pull/2276/files?diff=unified&w=0#diff-77c23b66e77777536c08734faf41242078eec3c5e0f21ea26a9c32e0e7c87e34L13-R13))
  - Add a maxLength attribute, a ngClass directive, and a span element to the input element for the report name ([link](https://github.com/fylein/fyle-mobile-app/pull/2276/files?diff=unified&w=0#diff-77c23b66e77777536c08734faf41242078eec3c5e0f21ea26a9c32e0e7c87e34L28-R37))
  - Define a error class in the SCSS file that adds a red border-bottom and a red color to the input and span elements when they have an error ([link](https://github.com/fylein/fyle-mobile-app/pull/2276/files?diff=unified&w=0#diff-8ca51b8c05b80a7daebe2b1db9af16f87699ee281eb30ac6264a1a35276aecafR38-R41), [link](https://github.com/fylein/fyle-mobile-app/pull/2276/files?diff=unified&w=0#diff-8ca51b8c05b80a7daebe2b1db9af16f87699ee281eb30ac6264a1a35276aecafR52-R56))
* Test the validation and limitation of report name length ([link](https://github.com/fylein/fyle-mobile-app/pull/2276/files?diff=unified&w=0#diff-38cd81416b5ece8a7192f0e3ffe1fd0082970ce71bb66d6b1f98c97eb6d58bc5R50-R57))
  - Add a test case that checks if the Save button is disabled when the report name length is greater than or equal to 250 ([link](https://github.com/fylein/fyle-mobile-app/pull/2276/files?diff=unified&w=0#diff-38cd81416b5ece8a7192f0e3ffe1fd0082970ce71bb66d6b1f98c97eb6d58bc5R50-R57))

## Clickup
https://app.clickup.com/t/85ztr0tg2

## Code Coverage
Please add code coverage here

### CODE REVIEW

File: src/app/fyle/my-view-report/edit-report-name-popover/edit-report-name-popover.component.html [hunk 1]
```diff
@@ -25,5 +25,14 @@
     <span> Report Name </span>
     <span class="edit-report-name--input-container__mandatory"> * </span>
   </div>
-  <input matInput [(ngModel)]="reportName" class="edit-report-name--input-container__input smartlook-show" />
+  <input
+    matInput
+    [(ngModel)]="reportName"
+    [maxLength]="250"
+    class="edit-report-name--input-container__input smartlook-show"
+    [ngClass]="{ 'edit-report-name--input-container__input--error': reportName?.trim().length >= 250 }"
```
Comment (Chethan-Fyle) [inline src/app/fyle/my-view-report/edit-report-name-popover/edit-report-name-popover.component.html:33]: If `maxLength` attribute is set to 250, how can the character count go above 250? 🤔

**Review by Chethan-Fyle — COMMENTED** [review]
One doubt.

File: src/app/fyle/my-view-report/edit-report-name-popover/edit-report-name-popover.component.html (hunk 1 above)
Comment (abhishek1234321) [reply src/app/fyle/my-view-report/edit-report-name-popover/edit-report-name-popover.component.html:33]: +1

**Review by abhishek1234321 — CHANGES_REQUESTED** [review]

File: src/app/fyle/my-view-report/edit-report-name-popover/edit-report-name-popover.component.html [hunk 2]
```diff
@@ -25,5 +25,10 @@
     <span> Report Name </span>
     <span class="edit-report-name--input-container__mandatory"> * </span>
   </div>
-  <input matInput [(ngModel)]="reportName" class="edit-report-name--input-container__input smartlook-show" />
+  <input
+    matInput
+    [(ngModel)]="reportName"
+    [maxLength]="250"
```
Comment (Dimple16) [inline src/app/fyle/my-view-report/edit-report-name-popover/edit-report-name-popover.component.html:31]: This is an edge case, I followed the Figma design and added the error message, and disabled the Save button
The previous changes are removed now
I have added the max length as 250, this change should do

**Review by Chethan-Fyle — APPROVED** [review]

**Review by abhishek1234321 — APPROVED** [review]

**Pull request merged**